brave/brave-core
# Fix upstream pdf ocr hanging on blank page

When `PdfAccessibilityTreeBuilder::AddRemainingAnnotations` processes ocr requests, blank page will have size 0 images_ per page, which makes the ocr request unprocessed.
In `PdfOcrHelper::ReceiveOcrResultsForImage`, it requires `remaining_page_count_` equals zero in order for
`PdfAccessibilityTree::OnOcrDataReceived` to be called and then also for `PdfOcrHelper::AreAllPagesOcred()` to be satisfied in order to set IDS_PDF_OCR_COMPLETED or IDS_PDF_OCR_NO_RESULT otherwise it would be stuck in IDS_PDF_OCR_IN_PROGRESS forever.

Page images info is passed into ax tree builder from PdfViewWebPlugin::PrepareAndSetAccessibilityPageInfo()-> chrome_pdf::GetAccessibilityInfo()->PDFiumEngine::GetImageInfo()-> PDFiumPage::GetImageInfo()->PDFiumPage::CalculateImages() and passed `AccessibilityPageObjects` into
PdfAccessibilityTree::SetAccessibilityPageInfo()-> PdfAccessibilityTree::AddPageContent()

So we create an empty image for blank page in
PDFiumPage::CalculateImages() to fix the issue.


Resolves https://github.com/brave/brave-browser/issues/41113



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Load any pdf that contains a blank page.
2. Open Leo and do summarize.
3. There should be response instead of waiting forever.
